### PR TITLE
Global ajax functions

### PR DESCRIPTION
**Pull request by somewebmedia**
Created one global function edd_ajax() that receives data from all other
functions and delivers the request. Each ajax call has it's own function
(edd_add_to_cart, edd_remove_from_cart, edd_update_quantity...), and
accepts it's properties, than calls edd_ajax(). Each of them accepts two
extra parameters, success and fail. They are the ajax callback functions
and are optional.

This way developers can use those functions without the need to know or
write their own ajax requests, so they can manipulate HTML on callback
in their own way. Default EDD button events and callbacks stayed the
same, but also go through the new functions.

<strong>!important this should be more properly tested!</strong>

### CODE REVIEW

**Comment by pippinsplugins** [comment]
@somewebmedia sorry about this getting lost.

I'd love to hear @easydigitaldownloads/core-devs opinions on this. There's some tremendous value in this for sure.

**Comment by spencerfinnell** [comment]
Any advantage to this over using `wp.ajax.post`or `wp.ajax.send`, which accepts success/error callbacks, and a few more options?

https://github.com/WordPress/WordPress/blob/master/wp-includes/js/wp-util.js#L72

**Comment by pippinsplugins** [comment]
@spencerfinnell It's more the functions that let developers easily add an item to the cart, remove a discount, etc, from JS.

I would like to see them use the `wp` functions though.

**Comment by spencerfinnell** [comment]
Yeah I like splitting out all the reusable stuff, but most of the things in `edd_ajax` are already handled by `wp.ajax.send` -- could still be used as a wrapper though to set a few options (especially the URL)

**Comment by somewebmedia** [comment]
@pippinsplugins @spencerfinnell I can try replacing jQuery ajax with wp's ajax and see how it goes.

**Comment by pippinsplugins** [comment]
Great. When you do, let's please open this as a new issue and a new PR (both issue and PR please).
